cli/src/cmd/forge/script/broadcast.rs

```diff
@@ -0,0 +1,266 @@
+                Ok(pending) => pending.await.map_err(|e| eyre::eyre!(e)),
+                Err(e) => Err(eyre::eyre!(e.to_string())),
+            }
+        }
+
+        let nonce = *legacy_or_1559.nonce().expect("no nonce");
+        let receipt = match broadcast(signer, legacy_or_1559).await {
+            Ok(Some(res)) => (res, nonce),
+
+            Ok(None) => {
+                // todo what if it has been actually sent
+                eyre::bail!("Failed to get transaction receipt?")
+            }
+            Err(e) => {
+                eyre::bail!("Aborting! A transaction failed to send: {:#?}", e)
+            }
+        };
+
+        Ok(receipt)
+    }
+
+    pub async fn handle_broadcastable_transactions(
+        &self,
+        target: &ArtifactId,
+        result: ScriptResult,
+        decoder: &mut CallTraceDecoder,
+        script_config: &ScriptConfig,
+    ) -> eyre::Result<()> {
+        if let Some(txs) = result.transactions {
+            if script_config.evm_opts.fork_url.is_some() {
+                if let Ok(gas_filled_txs) =
+                    self.execute_transactions(txs, script_config, decoder).await
+                {
+                    if !result.success {
+                        eyre::bail!("\nSIMULATION FAILED");
+                    } else {
+                        let txs = gas_filled_txs;
+                        let fork_url = self.evm_opts.fork_url.as_ref().unwrap().clone();
+
+                        let provider = get_http_provider(&fork_url);
+                        let chain = provider.get_chainid().await?.as_u64();
+
+                        let mut deployment_sequence = ScriptSequence::new(
+                            txs,
+                            &self.sig,
+                            target,
+                            &script_config.config,
+                            chain,
+                        )?;
+
+                        if self.broadcast {
+                            self.send_transactions(&mut deployment_sequence, &fork_url).await?;
+                        } else {
+                            println!("\nSIMULATION COMPLETE. To broadcast these transactions, add --broadcast and wallet configuration(s) to the previous command. See forge script --help for more.");
+                        }
+                    }
+                } else {
+                    eyre::bail!("One or more transactions failed when simulating the on-chain version. Check the trace by re-running with `-vvv`")
+                }
+            } else {
+                println!("\nIf you wish to simulate on-chain transactions pass a RPC URL.");
+            }
+        } else if self.broadcast {
+            eyre::bail!("No onchain transactions generated in script");
+        }
+        Ok(())
+    }
+
+    async fn wait_for_receipts(
+        &self,
+        tasks: Vec<impl futures::Future<Output = eyre::Result<(TransactionReceipt, U256)>>>,
```

Comment:
there's an argument to be had that this should be a `Stream` type that's a `FuturesOrdered` wrapper or something.